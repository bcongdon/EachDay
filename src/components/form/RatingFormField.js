import React from 'react'
import { Rating } from 'semantic-ui-react'
import { PropTypes } from 'prop-types'

function RatingFormField (props) {
  function handleChange (e, { rating, maxRating }) {
    props.input.onChange(rating)
  }
  const { input } = props
  return (
    <Rating rating={input.value} clearable={props.clearable} onRate={handleChange} maxRating={5} size='large' />
  )
}

RatingFormField.propTypes = {
  input: PropTypes.object.isRequired,
  clearable: PropTypes.bool
}

export default RatingFormField
