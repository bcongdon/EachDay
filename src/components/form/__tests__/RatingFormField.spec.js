import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Rating } from 'semantic-ui-react'
import RatingFormField from '../RatingFormField'

describe('RatingFormField component', () => {
  it('should render without rating if none provided', () => {
    var rating = mount(
      <RatingFormField
        clearable
        input={{value: null}}
        />
    )
    expect(rating.find(Rating).props().maxRating).toBe(5)
    expect(toJson(rating)).toMatchSnapshot()
  })

  it('should render if initial rating provided', () => {
    var rating = mount(
      <RatingFormField
        clearable
        input={{value: 3}}
        />
    )
    expect(rating.find(Rating.Icon).findWhere((n) => n.props().active).length).toBe(3)
    expect(toJson(rating)).toMatchSnapshot()
  })

  it('should handle onClick/handleChange correctly', () => {
    const onChange = jest.fn()
    var rating = mount(
      <RatingFormField
        clearable
        input={{value: 3, onChange}}
        />
    )
    rating.find(Rating.Icon).last().simulate('click')
    expect(onChange).toBeCalled()
    expect(onChange.mock.calls[0][0]).toBe(5)

    rating.setProps({input: {value: 5}})
    expect(rating.find(Rating.Icon).findWhere((n) => n.props().active).length).toBe(5)
    expect(toJson(rating)).toMatchSnapshot()
  })
})
