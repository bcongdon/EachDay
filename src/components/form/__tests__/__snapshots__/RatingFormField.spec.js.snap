// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingFormField component should handle onClick/handleChange correctly 1`] = `
<RatingFormField
  clearable={true}
  input={
    Object {
      "value": 5,
    }
  }
>
  <Rating
    clearable={true}
    maxRating={5}
    onRate={[Function]}
    rating={5}
    size="large"
  >
    <div
      className="ui large rating"
      onMouseLeave={[Function]}
      role="radiogroup"
    >
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={1}
        aria-setsize={5}
        index={0}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={1}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={2}
        aria-setsize={5}
        index={1}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={2}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={3}
        aria-setsize={5}
        index={2}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={3}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={4}
        aria-setsize={5}
        index={3}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={4}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={true}
        aria-posinset={5}
        aria-setsize={5}
        index={4}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={true}
          aria-posinset={5}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
    </div>
  </Rating>
</RatingFormField>
`;

exports[`RatingFormField component should render if initial rating provided 1`] = `
<RatingFormField
  clearable={true}
  input={
    Object {
      "value": 3,
    }
  }
>
  <Rating
    clearable={true}
    maxRating={5}
    onRate={[Function]}
    rating={3}
    size="large"
  >
    <div
      className="ui large rating"
      onMouseLeave={[Function]}
      role="radiogroup"
    >
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={1}
        aria-setsize={5}
        index={0}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={1}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={2}
        aria-setsize={5}
        index={1}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={2}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={true}
        aria-posinset={3}
        aria-setsize={5}
        index={2}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={true}
          aria-posinset={3}
          aria-setsize={5}
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={4}
        aria-setsize={5}
        index={3}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={4}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={5}
        aria-setsize={5}
        index={4}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={5}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
    </div>
  </Rating>
</RatingFormField>
`;

exports[`RatingFormField component should render without rating if none provided 1`] = `
<RatingFormField
  clearable={true}
  input={
    Object {
      "value": null,
    }
  }
>
  <Rating
    clearable={true}
    maxRating={5}
    onRate={[Function]}
    rating={null}
    size="large"
  >
    <div
      className="ui large rating"
      onMouseLeave={[Function]}
      role="radiogroup"
    >
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={1}
        aria-setsize={5}
        index={0}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={1}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={2}
        aria-setsize={5}
        index={1}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={2}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={3}
        aria-setsize={5}
        index={2}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={3}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={4}
        aria-setsize={5}
        index={3}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={4}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={5}
        aria-setsize={5}
        index={4}
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
      >
        <div
          aria-checked={false}
          aria-posinset={5}
          aria-setsize={5}
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
    </div>
  </Rating>
</RatingFormField>
`;
